FrontEnd/src/Pages/ControlPanel/PagesCP/profesores/Profesores.jsx

```diff
@@ -1,619 +1,161 @@
-                    {bodyInsertar}
-                </Modal>
-
-                <Modal open={modalEditar} onClose={() => abrirCerrarModalEditar()}>
-                    {bodyEditar}
-                </Modal>
+  const getProfesores2 = async () => {
+    await getProfesors().then((response) => response.json()).then((result) => {
+      setOriginalProfesorData(result);
+    });
+  };
 
-                <Modal open={modalEliminar} onClose={abrirCerrarModalEliminar}>
-                    {bodyEliminar}
-                </Modal>
+  const getClases = async () => {
+    await getClasses().then((response) => response.json()).then((result) => {
+      setDataClase(result);
+    });
+  };
 
-            </Box>
-        </div>
-    );
+  const getAllPeriodos = async () => {
+    await getPeriodos().then((response) => response.json()).then((result) => {
+      setDataPeriodo(result);
+    });
+  };
+
+  useEffect(() => {
+    getProfesores();
+    getProfesores2();
+    getAllPeriodos();
+    getClases();
+  }, []);
+
+  const abrirCerrarModal = () => {
+    if (openModal) {
+      setProfesorSeleccionado(profesorVacio);
+    }
+    setOpenModal(!openModal);
+  };
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setProfesorSeleccionado((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const seleccionarProfesor = (profesor, caso) => {
+    setProfesorSeleccionado(profesor);
+    setCurrentOperation(caso);
+    abrirCerrarModal();
+  };
+
+  const modalSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      if (currentOperation === 'Eliminar') {
+        await deleteProfesor({
+          _id: profesorSeleccionado._id,
+        });
+      } else if (currentOperation === 'Crear') {
+        await createProfesor({
+          nombre: profesorSeleccionado.nombre,
+          apellidos: profesorSeleccionado.apellidos,
+          matricula: profesorSeleccionado.matricula,
+          correo: profesorSeleccionado.correo,
+          fecha_de_nacimiento: profesorSeleccionado.fecha_de_nacimiento,
+          num_telefono: profesorSeleccionado.num_telefono,
+          num_cursos_impartidos: profesorSeleccionado.num_cursos_impartidos,
+          idUser: profesorSeleccionado.idUser,
+        });
+      } else {
+        await updateProfesor({
+          _id: profesorSeleccionado._id,
+          nombre: profesorSeleccionado.nombre,
+          apellidos: profesorSeleccionado.apellidos,
+          matricula: profesorSeleccionado.matricula,
+          correo: profesorSeleccionado.correo,
+          fecha_de_nacimiento: profesorSeleccionado.fecha_de_nacimiento,
+          num_telefono: profesorSeleccionado.num_telefono,
+          num_cursos_impartidos: profesorSeleccionado.num_cursos_impartidos,
+          idUser: profesorSeleccionado.idUser,
+        });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+    abrirCerrarModal();
+    getProfesores();
+  };
+
+  let array = [];
```

Comment:
I think that this could also be rewritten. I tried but there was a bug. Also will add it to the next PR.

Reply:
It's already on this PR.